portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,19 +25,51 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> messages = new ArrayList<>(Arrays.asList("Hello", "Bonjur", "Hola"));
+  private ArrayList<String> messages = new ArrayList<>();
 
+  /**
+   * Write to /data the messages ArrayList as a json string
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String json = convertToJson(messages);
-
     response.setContentType("text/html;");
     response.getWriter().println(json);
   }
 
+  /**
+   * Obtain the input from the comment form and add it to the messages ArrayList
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the comment input from the form.
+    String comment = getParameter(request, "text-input", /* default value= */ "");
+    messages.add(comment);
+
+    // Respond with the result.
+    response.setContentType("text/html;");
+    response.getWriter().println(convertToJson(messages));
+    response.sendRedirect("index.html#comments-container");
+  }
+
+  /**
+   * @return the ArrayList paramater converted to a json format
+   */
   public String convertToJson(ArrayList<String> messages) {
```

Comment:
convertMessageListToJson? Just make the functionality a bit more explicit.

Reply:
Sure!